components/board/genericlinux/board.go

```diff
@@ -39,21 +39,27 @@ func RegisterBoard(modelName string, gpioMappings map[int]GPIOBoardMapping, useP
 				conf resource.Config,
 				logger golog.Logger,
 			) (board.Board, error) {
-				return newBoard(ctx, conf, gpioMappings, usePeriphGpio, logger)
+				newConf, err := resource.NativeConfig[*Config](conf)
+				if err != nil {
+					return nil, err
+				}
+				return NewBoard(ctx, conf.ResourceName().AsNamed(), newConf, gpioMappings, usePeriphGpio, logger)
 			},
 		})
 }
 
-func newBoard(
+// NewBoard creates a new SysfsBoard.
+func NewBoard(
 	ctx context.Context,
-	conf resource.Config,
+	named resource.Named,
+	conf *Config,
 	gpioMappings map[int]GPIOBoardMapping,
 	usePeriphGpio bool,
 	logger golog.Logger,
-) (board.Board, error) {
+) (*SysfsBoard, error) {
```

Comment:
was running into weird casting errors trying this but was wondering whats the advantage of doing this since the NewBoard function creates the new object as a genericlinux type board anyway

Reply:
We want all our creator functions in drivers to return the go types so that a) the code is maintainable, b) anything that accesses the constructors over the line in the future can rely on a specific interface attached to an API being returned from the creator function. Otherwise, maintainability is harder.